lib/src/parsed_path.dart

```diff
@@ -161,18 +161,40 @@ class ParsedPath {
     return builder.toString();
   }
 
+  /// Returns k-th last index of the `character` in the `path`.
+  ///
+  /// If `k` exceeds the count of `character`s in `path`, the left most index
+  /// of the `character` is returned.
+  int _kthLastIndexOf(String path, String character, int k) {
+    var count = 0, leftMostIndexedCharacter = 0;
+    for (var index = path.length - 1; index >= 0; --index) {
+      if (path[index] == character) {
+        leftMostIndexedCharacter = index;
+        ++count;
+        if (count == k) {
+          return index;
+        }
+      }
+    }
+    return leftMostIndexedCharacter;
+  }
+
   /// Splits the last non-empty part of the path into a `[basename, extension`]
   /// pair.
   ///
+  /// Takes an optional parameter `level` which makes possible to return
+  /// multiple extensions having `level` number of dots. If `level` exceeds the
+  /// number of dots, the path is splitted into the left most dot.
+  ///
   /// Returns a two-element list. The first is the name of the file without any
   /// extension. The second is the extension or "" if it has none.
-  List<String> _splitExtension() {
+  List<String> _splitExtension([int level = 1]) {
```

Comment:
We should check that it's non-null. After the null safe migration we'll be OK, but until then we should guard against it

```suggestion
  List<String> _splitExtension([int level = 1]) {
    ArgumentError.checkNotNull(level, 'level');
```

I'm also not sure if we need a `RangeError` check here. At a minimum `RangeError.checkNotNegative(level, 'level');`

Depending on the behavior of passing `0` we might instead need:

```dart
if(level < 0) throw RangeError(level, 1);
```

Reply:
Looks like `checkNotNull` was introduced in `2.1`, so CI failing for dart `2.0`. Can we use [notNull](https://api.dart.dev/stable/1.22.0/dart-core/ArgumentError/ArgumentError.notNull.html) instead?

Reply:
That would be fine for now.